src/subdomains/supporting/payin/services/payin.service.ts

```diff
@@ -183,7 +185,47 @@ export class PayInService {
     }
   }
 
-  private async returnPayIns(): Promise<void> {
+  private async checkOutputConfirmations(): Promise<void> {
+    const payIns = await this.payInRepository.find({
+      where: { status: PayInStatus.FORWARDED },
+      take: 10000,
+    });
+
+    if (payIns.length === 0) return;
+
+    const groups = this.groupByStrategies(payIns, (a) => this.sendStrategyRegistry.getSendStrategy(a));
+
+    for (const group of groups.entries()) {
+      try {
+        const strategy = group[0];
+        await strategy.checkConfirmations(group[1], 'Output');
+      } catch {
+        continue;
+      }
+    }
+  }
+
+  private async checkReturnConfirmations(): Promise<void> {
+    const payIns = await this.payInRepository.find({
+      where: { status: PayInStatus.RETURNED },
+      take: 10000,
+    });
+
+    if (payIns.length === 0) return;
+
+    const groups = this.groupByStrategies(payIns, (a) => this.sendStrategyRegistry.getSendStrategy(a));
+
+    for (const group of groups.entries()) {
+      try {
+        const strategy = group[0];
+        await strategy.checkConfirmations(group[1], 'Return');
+      } catch {
+        continue;
+      }
+    }
+  }
+
+  private async confirmReturnedPayIns(): Promise<void> {
     const payIns = await this.payInRepository.findBy({
```

Comment:
Weird method name? Nothing is confirmed here

Reply:
Hm but then the status `ReturnConfirmed` a bit strange or? 😅 what would you call it?